### PR TITLE
doc: Add ORDER BY documentation

### PR DESCRIPTION
**Pull request by Gashmob**
Part of [story #39685 Order artifact according to a field](https://tuleap.net/plugins/tracker/?aid=39685)

### CODE REVIEW

File: languages/en/user-guide/tql.rst [hunk 1]
```diff
@@ -330,13 +334,22 @@ If only the tracker condition is provided, then match the trackers from current
 
 To provide both condition, you can use ``AND`` between them. There is no restriction for the order of the conditions.
 
+TQL ``ORDER BY`` syntax allow you to sort artifacts according to a field. The allowed fields are the same as for the condition with the same restrictions.
```
Comment (Hyzual) [inline languages/en/user-guide/tql.rst:337]: issue: It's not exactly the same fields: sorting is only allowed on list fields with a single possible value (selectbox and radio buttons, NOT open lists, multi-selectbox or checkbox)

File: languages/en/user-guide/tql.rst [hunk 2]
```diff
@@ -330,13 +334,22 @@ If only the tracker condition is provided, then match the trackers from current
...
+
```
Comment (Hyzual) [inline languages/en/user-guide/tql.rst:338]: suggestion: we should also mention that ordering does not allow mixing date and datetime fields

File: languages/en/user-guide/tql.rst [hunk 3]
```diff
@@ -330,13 +334,22 @@ If only the tracker condition is provided, then match the trackers from current
...
+You must provide the direction of the ordering:
+ * ``ASC`` or ``ASCENDING`` from smallest to largest
+ * ``DESC`` or ``DESCENDING`` from largest to smallest
+
+If you do not provide an ``ORDER BY`` to your query, it will default to ``ORDER BY @id DESCENDING``.
```
Comment (Hyzual) [inline languages/en/user-guide/tql.rst:343]: praise: :+1:

**Review by Hyzual — CHANGES_REQUESTED** [review]

File: languages/en/user-guide/tql.rst (hunk 1 above)
Comment (Gashmob) [reply languages/en/user-guide/tql.rst]: I completely forgot this point

File: languages/en/user-guide/tql.rst (hunk 2 above)
Comment (Gashmob) [reply languages/en/user-guide/tql.rst:341]: Done

**Review by Hyzual — CHANGES_REQUESTED** [review]

File: languages/en/user-guide/tql.rst [hunk 4]
```diff
@@ -330,13 +334,25 @@ If only the tracker condition is provided, then match the trackers from current
 
 To provide both condition, you can use ``AND`` between them. There is no restriction for the order of the conditions.
 
+TQL ``ORDER BY`` syntax allow you to sort artifacts according to a field.
+
+There is some restrictions on the fields you can use. As for the condition their types must be compatible, date and date with time fields are not compatible. 
+For list field only selectbox and radio buttons are allowed. All the ``@`` fields are allowed.
```
[the remark this answers is not in the document]
Comment (Gashmob) [reply languages/en/user-guide/tql.rst:340]: Done

File: languages/en/user-guide/tql.rst [hunk 5]
```diff
@@ -330,13 +334,25 @@ If only the tracker condition is provided, then match the trackers from current
 
 To provide both condition, you can use ``AND`` between them. There is no restriction for the order of the conditions.
 
+TQL ``ORDER BY`` syntax allow you to sort artifacts according to a field.
+
+There is some restrictions on the fields you can use. As for the condition their types must be compatible, date and date with time fields are not compatible.
```
[the remark this answers is not in the document]
Comment (Gashmob) [reply languages/en/user-guide/tql.rst]: Done

File: languages/en/user-guide/tql.rst [hunk 6]
```diff
@@ -330,13 +334,25 @@ If only the tracker condition is provided, then match the trackers from current
 
 To provide both condition, you can use ``AND`` between them. There is no restriction for the order of the conditions.
 
+TQL ``ORDER BY`` syntax allow you to sort artifacts according to a field.
```
[the remark this answers is not in the document]
Comment (Gashmob) [reply languages/en/user-guide/tql.rst]: Done

File: languages/en/user-guide/tql.rst [hunk 7]
```diff
@@ -330,13 +334,28 @@ If only the tracker condition is provided, then match the trackers from current
...
+TQL ``ORDER BY`` syntax allows you to sort artifacts on a single field. Ordering on more than one criteria is not possible.
+
+There are some restrictions on the fields you can use. As for the condition (``WHERE``) their types must be compatible. Date and date with time fields are not compatible together.
+For list fields, only selectbox and radio buttons are allowed. All the ``@`` fields and semantics valid in conditions (``WHERE``) are allowed.
+
+You must provide the direction of the ordering:
+ * ``ASC`` or ``ASCENDING`` from smallest to largest
+ * ``DESC`` or ``DESCENDING`` from largest to smallest
+
+User list and user group list fields are compared on displayed value. For user lists it means that result depends on your preference on user display (login vs realname).
```
Comment (Hyzual) [inline languages/en/user-guide/tql.rst:346]:
```suggestion
User list and user group list fields are compared on displayed value. For user lists, it means that the ordering depends on your preference on user display (login, real name or both).
```

File: languages/en/user-guide/tql.rst [hunk 8]
```diff
@@ -330,13 +334,28 @@ If only the tracker condition is provided, then match the trackers from current
...
+
+There are some restrictions on the fields you can use. As for the condition (``WHERE``) their types must be compatible. Date and date with time fields are not compatible together.
+For list fields, only selectbox and radio buttons are allowed. All the ``@`` fields and semantics valid in conditions (``WHERE``) are allowed.
+
+You must provide the direction of the ordering:
+ * ``ASC`` or ``ASCENDING`` from smallest to largest
+ * ``DESC`` or ``DESCENDING`` from largest to smallest
+
+User list and user group list fields are compared on displayed value. For user lists it means that result depends on your preference on user display (login vs realname). 
+For user groups like project members or project admins, the sort is done on their translated name and so result depends on your language.
```
Comment (Hyzual) [inline languages/en/user-guide/tql.rst:347]:
```suggestion
For user groups like Project members or Project administrators, the sort is done on their translated name and so the ordering depends on your language.
```

**Review by Hyzual — APPROVED** [review]
Remaining comments are optional

File: languages/en/user-guide/tql.rst (hunk 7 above)
Comment (Gashmob) [reply languages/en/user-guide/tql.rst:346]: Done

File: languages/en/user-guide/tql.rst (hunk 8 above)
Comment (Gashmob) [reply languages/en/user-guide/tql.rst]: Done

**Pull request merged**